README.md

```diff
@@ -15,6 +15,42 @@ In other words, by using this handler with Sceptre, _you skip ever using `cdk de
 `launch`, `validate`, `generate`, and `diff` (along with all the rest)!
 
+## Why would you want to use CDK with Sceptre? 
+**Aren't those two ways to do the same thing?**
+
+CDK and Sceptre really specialize in two different things, though there's definitely overlap.
+
+CDK's whole philosophy is that developers should be able to write code in a language they know and
+have that compile into an _artifact_ that defines cloud infrastructure resources. Of course, that
+"artifact" is ultimately a CloudFormation template. **CDK's specialty, thus, is really in _programmatic_
+template generation with sane defaults and constructs that simplify infrastructure configuration.** It
+hides a lot of the "magic" behind the scenes so developers can simply prescribe what they want. CDK
+also has deployment mechanisms to deploy that infrastructure, but they aren't really CDK's strength.
+
+90% of everything CDK does is at "compile-time", when the templates are being rendered. It has a very
+limited ability to execute custom code at deployment time, and never really _between_ the deployments 
+of custom stacks and resources. Furthermore, its way of wiring together stacks uses CloudFormation 
+exported outputs and imports, which are rather rigid and can have a host of unintended consequences, 
+especially when values might change.
+
+In contrast, Sceptre is a _deployment orchestration_ tool. It excels in deploying entire environments
+of stacks, "wiring them together" using powerful (and easily customizable) resolvers, hooks, and
+template handlers. **Sceptre is "template agnostic"**, supporting YAML and JSON CloudFormation
+templates, even augmented with Jinja2 execution logic. Furthermore, out-of-the-box, Sceptre supports 
+_any custom Python code to generate templates_, with the only requirement being that it needs to return
+a string.
+
+_So why use CDK and Sceptre together?_ Because CDK provides excellent template generation capabilities
+and Sceptre will gladly use those. Furthermore, Sceptre has the ability to easily wire together an
+entire environment _regardless of how that environment's CloudFormation templates are generated._ Thus,
+Sceptre will happily (and fairly intuitively) deploy (and "wire together") stacks developed in vanilla
+CloudFormation YAML/JSON, templates augmented with Jinja2, Troposphere-generated templates, 
+AWS SAM templates, and CDK constructs. They can be deployed as a coherent environment that 
+interoperates, deployed with insight into the various dependencies between stacks. Furthermore, 
+using Sceptre's powerful hooks, you can execute customized pre-deployment and post-deployment code 
+to prepare the way for or clean up after a given stack deployment. This is powerful and not something 
+CDK provides a means to accomplish.
+
 ## How to install sceptre-cdk-handler
 
 1) Install the Sceptre CDK handler using `pip install sceptre-cdk-handler`
```

Comment:
step #3 is only needed for `bootstrapped` mode, correct?

is step #2 even needed at all?  seem optional to me since this handler only needs the python library dependencies?

Reply:
No, Step 3 is always necessary. We still have the possibility of CDK assets. The difference is that the `bootstrapped` deployment_type uses the bootstrap stack's bucket, ECR repo, and IAM roles to push those cdk assets, whereas the `bootstrapless` deployment_type uses whatever bucket, repo (and optionally iam roles) we specify in the `bootstrapless_config`. In other words, the `deployment_type` is the "_as whom_" and the "_where to_" related to the assets. But assets will likely need to get pushed regardless.

Reply:
> is step https://github.com/Sceptre/sceptre-cdk-handler/pull/2 even needed at all? seem optional to me since this handler only needs the python library dependencies?

I think you might actually be right! I think you used to need CDK installed to do the synth for the CDK v1. But the new v2 seems to not need it. But I might be wrong. @X-Guardian , do you happen to have any context on this? I retained it because you originally had this in the instructions. But I just tested using aws-cdk-lib installed in a clean python virtual environment without access to cdk and I was able to synth the template without installing CDK.

Reply:
To be safe, I'm going to leave this as it is. I think there's risk that there are use cases that would require CDK to be installed.